content/automate/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogiteminitialization.rb

```diff
@@ -102,27 +120,25 @@ def set_all_vm_name_attrs(prov, new_vm_name)
 end
 
 def service_item_dialog_values(dialogs_options_hash)
-  merged_options_hash = Hash.new { |h, k| h[k] = {} }
   provision_index = determine_provision_index
 
-  if dialogs_options_hash[0].nil?
-    merged_options_hash = dialogs_options_hash[provision_index] || {}
-  else
-    merged_options_hash = dialogs_options_hash[0].merge(dialogs_options_hash[provision_index] || {})
-  end
+  merged_options_hash = if dialogs_options_hash[0].nil?
+                          dialogs_options_hash[provision_index] || {}
+                        else
+                          dialogs_options_hash[0].merge(dialogs_options_hash[provision_index] || {})
+                        end
   merged_options_hash
 end
 
 def service_item_tag_values(dialogs_tags_hash)
-  merged_tags_hash         = Hash.new { |h, k| h[k] = {} }
   provision_index = determine_provision_index
 
   # merge dialog_tag_0 stuff with current build
-  if dialogs_tags_hash[0].nil?
-    merged_tags_hash = dialogs_tags_hash[provision_index] || {}
-  else
-    merged_tags_hash = dialogs_tags_hash[0].merge(dialogs_tags_hash[provision_index] || {})
-  end
+  merged_tags_hash = if dialogs_tags_hash[0].nil?
```

Comment:
@billfitzgerald0120 we dont need to use the merged_tags_hash, last assigned value in the if block will get returned